File: frontend/src/modules/data/types/data_platform.ts
export interface DatasetItem {
  id: string;
  name: string;
  schemaJson: string;
  storageType: string;
  rowCount: number;
  sizeBytes: number;
  createdAt: string;
}

export interface DataPipelineItem {
  id: string;
  name: string;
  schedule: string;
  status: string;
  createdAt: string;
}

export interface CatalogEntryItem {
  id: string;
  name: string;
  description?: string;
  owner: string;
  tagsJson: string;
  piiFlag: boolean;
  createdAt: string;
}

export interface QualityMetricItem {
  id: string;
  datasetId: string;
  completenessScore: number;
  validityScore: number;
  freshnessSec: number;
  status: string;
  checkedAt: string;
}

export interface SemanticMetricItem {
  id: string;
  name: string;
  measureSql: string;
  dimensionName: string;
  category: string;
  createdAt: string;
}

File: backend/app/schemas/data_platform.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class DatasetResponse(BaseModel):
    id: str
    name: str
    dataset_schema_json: str
    storage_type: str
    row_count: int
    size_bytes: int
    created_at: datetime

    class Config:
        from_attributes = True

class DataPipelineResponse(BaseModel):
    id: str
    name: str
    schedule: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class IngestionRequest(BaseModel):
    source_type: str = "PostgreSQL"
    source_table: str = "orders"
    destination_dataset: str = "lakehouse_sales_orders"
    mode: str = "CDC" # CDC, Batch, Streaming

class QueryRequest(BaseModel):
    sql_text: str = "SELECT * FROM lakehouse_sales_orders WHERE status = 'completed' LIMIT 100;"

class QueryResponse(BaseModel):
    query_id: str
    sql_text: str
    execution_time_ms: int
    rows_returned: int
    columns: List[str]
    data: List[Dict[str, Any]]
    status: str

class CatalogEntryResponse(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    owner: str
    tags_json: str
    pii_flag: bool
    created_at: datetime

    class Config:
        from_attributes = True

class QualityMetricResponse(BaseModel):
    id: str
    dataset_id: str
    completeness_score: float
    validity_score: float
    freshness_sec: int
    status: str
    checked_at: datetime

    class Config:
        from_attributes = True

class SemanticMetricResponse(BaseModel):
    id: str
    name: str
    measure_sql: str
    dimension_name: str
    category: str
    created_at: datetime

    class Config:
        from_attributes = True
